unblob/processing.py

```diff
@@ -345,15 +347,26 @@ def _process_chunks(
 
         for chunk in unknown_chunks:
             carved_unknown_path = carve_unknown_chunk(self.carve_dir, file, chunk)
-            self._calculate_entropy(carved_unknown_path)
-            self.result.add_report(chunk.as_report())
+            entropy = self._calculate_entropy(carved_unknown_path)
+            self.result.add_report(chunk.as_report(entropy=entropy))
```

Comment:
I don't understand why you add the `EntropyReport` to the results in two different ways. One line 336-337 you check if entropy is set and then add it as is, on this line you don't check if it is set and add it using `chunk.as_report()`.

Reply:
yes, because the previous instance is not chunk related, the entropy report is added to the result directly *if* it was calculated, while in the current case the entropy report is part of the chunk-report (or not), but the chunk report can be added unconditionally.